### PR TITLE
fix: correctly type the size property

### LINKED ISSUE #220: Incorrect @Input size typing when strictTemplates is enabled
**Describe the bug**
AOT compilation build under Angular 9 with `angularCompilerOptions.strictTemplates: true` in `tsconfig.app.json` generates a `error TS2322: Type 'string' is not assignable to type 'number'` when binding `<as-split-area [size]="'*'">`, as the `@Input size` is typed as `number | null` in[ the source code](https://github.com/bertrandg/angular-split/blob/master/projects/angular-split/src/lib/directive/splitArea.directive.ts#L28), though the [documentation](https://bertrandg.github.io/angular-split/#/documentation) clearly states it can also be bound to `'*'` in pixel mode.

### PR DESCRIPTION
**Pull request by beeman**
Fix #220

### CODE REVIEW

File: projects/angular-split/src/lib/interface.ts [hunk 1]
```diff
@@ -5,13 +5,15 @@ export interface IPoint {
   y: number
 }
 
+export type IAreaSize = number | '*'
```
Comment (lf-novelt) [inline projects/angular-split/src/lib/interface.ts:8]: weird to prefix a type (not an interface) with the capital `I` prefix that means Interface :P

Comment (beeman) [reply projects/angular-split/src/lib/interface.ts:8]: Good point @lf-novelt - I'm not a fan of the I for interfaces either, it's more a matter of keeping consistency.

**Comment by lf-novelt** [comment]
Thanks for the PR.
Can you set the description of the first post to the following please?
Fix #220

**Comment by 108174** [comment]
Has this been merged yet? Still having the issue; using the $any('*') workaround for now

**Comment by beeman** [comment]
> Has this been merged yet? Still having the issue; using the $any('*') workaround for now

Sorry, little time for maintenance currently. I will see if I can cut a release and include this change.

**Comment by KyDenZ** [comment]
Can you merge this modification?

**Comment by zbagley** [comment]
Bump

**Comment by laggage** [comment]
Is there a plan to release this PR?

**Comment by Harpush** [comment]
@beeman The PR seems good... Any reason to wait with it?

**Comment by wf9a5m75** [comment]
For the people who are waiting this PR. This PR is still waiting for some reasons.
Meanwhile, you can do with `[size]="$any('*')`.

For example:

```ts
<as-split
  [direction]="'horizontal'"
  [unit]="'pixel'"
>
  
   <as-split-area
     [size]="280"
     [maxSize]="280">
    A
   </as-split-area>
   
   <as-split-area
     [size]="$any('*')">
    B
   </as-split-area>

</as-split>
```

**Pull request closed without merging**